lfs/ssh.go

```diff
@@ -38,3 +47,41 @@ func sshAuthenticate(endpoint Endpoint, operation, oid string) (sshAuthResponse,
 
 	return res, err
 }
+
+// Return the executable name for ssh on this machine and the base args
+// Base args includes port settings, user/host, everything pre the command to execute
+func sshGetExeAndArgs(endpoint Endpoint) (exe string, baseargs []string) {
+	if len(endpoint.SshUserAndHost) == 0 {
+		return "", nil
+	}
+
+	ssh := Config.Getenv("GIT_SSH")
+	basessh := filepath.Base(ssh)
+	// Strip extension for easier comparison
+	if ext := filepath.Ext(basessh); len(ext) > 0 {
+		basessh = basessh[:len(basessh)-len(ext)]
+	}
+	isPlink := strings.EqualFold(basessh, "plink")
+	isTortoise := strings.EqualFold(basessh, "tortoiseplink")
+	if ssh == "" {
+		ssh = "ssh"
+	}
```

Comment:
I like it. I think the workflow is clearer. @sinbad: Thoughts? I can just make this change after merging.